File: app/src/main/java/com/jiahaoliuliu/googlemapsroute/MainComponent.java
package com.jiahaoliuliu.googlemapsroute;

import com.jiahaoliuliu.datalayer.RepositoryModule;
import com.jiahaoliuliu.networklayer.NetworkModule;

import javax.inject.Singleton;

import dagger.Component;

@Component(modules = {ApplicationModule.class, RepositoryModule.class, NetworkModule.class})

@Singleton
public interface MainComponent {
    void inject(OriginFragment originFragment);
}

File: app/src/main/java/com/jiahaoliuliu/googlemapsroute/MainComponent.kt
package com.jiahaoliuliu.googlemapsroute

import com.jiahaoliuliu.datalayer.RepositoryModule
import com.jiahaoliuliu.networklayer.NetworkModule
import dagger.Component
import javax.inject.Singleton

@Component(modules = [ApplicationModule::class, RepositoryModule::class, NetworkModule::class])
@Singleton
interface MainComponent {
    fun inject(originFragment: OriginFragment?)
    fun inject(destinationFragment: DestinationFragment?)
    fun inject(locationSearchFragment: LocationSearchFragment?)
}